controller/src/main/java/io/pravega/controller/server/security/auth/LegacyAuthorizationResourceImpl.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.security.auth;
+
+import io.pravega.common.Exceptions;
+
+/**
+ * A legacy implementation that constructs resource strings in the old format.
+ */
+public final class LegacyAuthorizationResourceImpl implements AuthorizationResource {
+
+    public String ofScopes() {
+        return "/";
+    }
+
+    public String ofScope(String scopeName) {
+        Exceptions.checkNotNullOrEmpty(scopeName, "scopeName");
+        return scopeName;
+    }
+
+    public String ofStreamsInScope(String scopeName) {
+        return Exceptions.checkNotNullOrEmpty(scopeName, "scopeName");
+    }
+
+    public String ofStreamInScope(String scopeName, String streamName) {
+        Exceptions.checkNotNullOrEmpty(streamName, "streamName");
+        return String.format("%s/%s", ofStreamsInScope(scopeName), streamName);
```

Comment:
same here and elsewhere

Reply:
Specifying the format directly makes it easier to understand what's happening here, IMO.